src/components/LoginModal.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react'
+import { Button, Dimmer, Grid, Header, Loader, Modal } from 'semantic-ui-react'
+import SignInForm from './SignInForm'
+import SignUpForm from './SignUpForm'
+
+export class LoginModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      open: this.props.open,
+      loading: false
+    }
+  }
+  onSignInSubmit(username, password) {
+    this.setState({ loading: true })
+    return this.props.onSignIn(username, password)
+  }
+  onSignUpSubmit(username, password) {
+    this.setState({ loading: true })
+    return new Promise((res, rej) => setTimeout(res, 2000))
+  }
+  onError() {
+    this.setState({ loading: false })
+  }
+  onClose() {
+    this.setState({ open: false })
+  }
+  onOpen() {
```

Comment:
That might become a bit tricky (but not impossible). I would rather leave that to another issue.